### PR TITLE
Add globalization

### PR DESCRIPTION
**Pull request by 0candy**
https://github.com/strongloop/loopback/issues/2422

### CODE REVIEW

**Comment by 0candy** [comment]
@jannyHou or @superkhau PTAL. Thanks.

File: common/models/data-source-definition.js [hunk 1]
```diff
@@ -106,8 +109,8 @@ function ready(DataSourceDefinition) {
       // We have to build a useful error message ourselves
 
       return cb(
-        new Error('Cannot connect to the data source.' +
-          ' Ensure the configuration is valid and the connector is installed.'));
+        new Error(g.t('Cannot connect to the data source.' +
```
Comment (jannyHou) [inline common/models/data-source-definition.js]: `g.f`

File: common/models/config-file.js [hunk 2]
```diff
@@ -121,7 +124,7 @@ module.exports = function(ConfigFile) {
 
   ConfigFile.prototype.remove = function(cb) {
     var configFile = this;
-    if (!this.path) return cb(new Error('no path specified'));
+    if (!this.path) return cb(new Error(g.t('no path specified')));
```
Comment (jannyHou) [inline common/models/config-file.js]: `g.f`

File: common/models/config-file.js [hunk 3]
```diff
@@ -64,7 +67,7 @@ module.exports = function(ConfigFile) {
 
   ConfigFile.prototype.load = function(cb) {
     var configFile = this;
-    if (!this.path) return cb(new Error('no path specified'));
+    if (!this.path) return cb(new Error(g.t('no path specified')));
```
Comment (jannyHou) [inline common/models/config-file.js]: `g.f`

File: bin/setup-mysql.js [hunk 4]
```diff
@@ -19,7 +22,7 @@ var connection;
 async.series([
   function setupConnection(next) {
     read({
-      prompt: 'Enter password for MySQL root user:',
+      prompt: g.t('Enter password for MySQL root user:'),
```
Comment (jannyHou) [inline bin/setup-mysql.js]: `g.f`

File: bin/datasource-invoke.js [hunk 5]
```diff
@@ -36,8 +39,8 @@ function invoke(msg, cb) {
   var cbMsg = {};
   var app, ds;
 
-  assert(dataSourceName, 'dataSourceName is required');
-  assert(methodName, 'methodName is required');
+  assert(dataSourceName, g.t('dataSourceName is required'));
```
Comment (jannyHou) [inline bin/datasource-invoke.js]: `g.f` and next line.

**Comment by jannyHou** [comment]
@0candy Only some reminders that change `g.t` to `g.f`, they will be folded after change. Other than that LGTM 👍

File: bin/setup-mysql.js [hunk 6]
```diff
@@ -34,11 +37,11 @@ async.series([
     });
   },
   function createDatabase(next) {
-    console.log('Creating database', DATABASE);
+    console.log(g.f('Creating database %s', DATABASE));
     connection.query('CREATE DATABASE IF NOT EXISTS ' + DATABASE, next);
   },
   function createUser(next) {
-    console.log('Creating user %s with password %s', USER, PASSWORD);
+    console.log(g.f('Creating user %s with password %s', USER, PASSWORD));
```
Comment (gunjpan) [inline bin/setup-mysql.js]: I see inconsistency between using `console.log` or `g.log`

Comment (superkhau) [inline bin/setup-mysql.js]: :+1: We should allow either but probably recommend g.log in globalization.md.

**Pull request merged**